src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/HandleCollector.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices
+{
+    public static class HandleCollectorTests
+    {
+        private const int LowLimitSize = 20;
+        private const int HighLimitSize = 100000;
+
+        [Fact]
+        public static void NegativeInitialThresholdCtor()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new HandleCollector("NegativeInitial", -1));
+        }
+
+        [Fact]
+        public static void NegateMaximumThresholdCtor()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new HandleCollector("NegativeMax", 0, -1));
+        }
+
+        [Fact]
+        public static void InitialGreaterThanMaxThresholdCtor()
+        {
+            Assert.Throws<ArgumentException>(() => new HandleCollector("InitialGreaterThanMax", 100, 1));
+        }
+
+        [Fact]
+        public static void SimplePropertyValidation()
+        {
+            int initial = 10;
+            int max = 20;
+            HandleCollector collector = new HandleCollector(null, initial, max);
+
+            Assert.Equal(0, collector.Count);
+            Assert.Equal(initial, collector.InitialThreshold);
+            Assert.Equal(max, collector.MaximumThreshold);
```

Comment:
Yep, I think it's completely pointless actually... the only thing it does is compare the initialThreshold parameter against it and throws if initial > max.

That said, it's part of the contract that already shipped so we can't really do much about it, I think.